index.bs

```diff
@@ -575,79 +575,225 @@ urlPrefix: https://w3c.github.io/FileAPI/#dfn-; type: dfn;
+
+					1. Let |blobData| be a [=Blob=] created using |representation|'s [=data=] and with its {{Blob/type}} set to |mimeType|, [=serialize a MIME type|serialized=].
 
-		enum PresentationStyle { "unspecified", "inline", "attachment" };
+					1. Let |p| be [=a new promise=] in |realm|.
 
-		dictionary ClipboardItemOptions {
-		  PresentationStyle presentationStyle = "unspecified";
+					1. Resolve |p| with |blobData|.
+
+					1. Return |p|.
+
+			1. Return [=a promise rejected with=] {{"NotFoundError"}} DOMException in |realm|.
+
+	<h3 id="clipboard-interface">Clipboard Interface</h3>
+
+		<pre class="idl" data-highlight="webidl">
+		typedef sequence&lt;ClipboardItem&gt; ClipboardItems;
+
+		[SecureContext, Exposed=Window]
+		interface Clipboard : EventTarget {
+			Promise&lt;ClipboardItems&gt; read();
+			Promise&lt;DOMString&gt; readText();
+			Promise&lt;undefined&gt; write(ClipboardItems data);
+			Promise&lt;undefined&gt; writeText(DOMString data);
 		};
 		</pre>
 
-		<div id="clipboard-idl" dfn-for="Clipboard">
+		A <dfn>Clipboard</dfn> may contain <dfn>Clipboard Items</dfn> which has one or more [=clipboard item=]s. Multiple [=clipboard item=]s are only supported on platforms such as iOS/iPadOS.
+		A web author needs to create a |data| which is a [=Clipboard items=] object in order to write content to [=system clipboard=] using the {{Clipboard/write(data)}} method.
+		{{Clipboard/read()}} returns a [=Promise=] to [=Clipboard items=] object that represents contents of [=system clipboard=].
+		A [=clipboard item=] can be read from [=clipboard items=].
+
+		The <dfn>clipboard task source</dfn> is triggered in response to reading or writing of [=system clipboard data=].
 
+		<div id="clipboard-idl" dfn-for="Clipboard">
 		<div class="algorithm" data-algorithm="clipboard-read">
 		<h4 method for="Clipboard">read()</h4>
 		The {{Clipboard/read()}} method must run these steps:
 
-			1. Let |p| be a new [=Promise=].
+			1. Let |realm| be [=this=]'s [=relevant realm=].
+
+			1. Let |p| be [=a new promise=] in |realm|.
 
 			1. Run the following steps [=in parallel=]:
 
-				1. Let |r| be the result of running [=check clipboard read permission=] [=in parallel=]
+				1. Let |r| be the result of running [=check clipboard read permission=].
+
+					Note: Clipboard permission is not supported on Safari. However, the read() method must be called inside
+					a user gesture event and the user must select the paste option from the native context menu that pops up
+					when read() is called from JS, otherwise, the promise will be rejected.
+
+				1. If |r| is not "granted", then [=Queue a global task=] on the [=permission task source=], given |realm|'s [=Realm/global object=], to perform the below step:
+				
+					1. Reject |p| with {{"NotAllowedError"}} DOMException in |realm|.
 
-				1. If |r| is not "granted", then reject |p| with a "NotAllowedError" DOMException
+					1. Abort all steps.
```

Comment:
This is confusing, since you can't queue a task to abort steps happening in parallel.

I think the phrasing you want is something like:

> 1. If r is not granted, then:
>    1. Queue a global task ... to reject p with ...
>    1. Abort these steps.

Reply:
Done.